src/c-opentimelineio/tests/OTIOCompositionTests.cpp

```diff
@@ -929,4 +929,63 @@ TEST_F(OTIOTrackTests, DeleteParentContainerTest)
     sq                  = NULL;
     Composition* parent = Composable_parent((Composable*) it);
     EXPECT_EQ(parent, nullptr);
+}
+
+TEST_F(OTIOTrackTests, TransactionalTest)
+{
+    Item*            item        = Item_create(NULL, NULL, NULL, NULL, NULL);
+    Track*           trackA      = Track_create(NULL, NULL, NULL, NULL);
+    Track*           trackB      = Track_create(NULL, NULL, NULL, NULL);
+    OTIOErrorStatus* errorStatus = OTIOErrorStatus_create();
+    Item*            item1       = Item_create(NULL, NULL, NULL, NULL, NULL);
+    Item*            item2       = Item_create(NULL, NULL, NULL, NULL, NULL);
+    Item*            item3       = Item_create(NULL, NULL, NULL, NULL, NULL);
+
+    SerializableObject* itemClone1 =
+        SerializableObject_clone((SerializableObject*) item, errorStatus);
+
+    SerializableObject* itemClone2 =
+        SerializableObject_clone((SerializableObject*) item, errorStatus);
+    // TODO segfault here
```

Comment:
Again getting a segfault here.
Tried to debug using gdb (don't have a lot of experience with this). 

It seems to be occuring here: https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/eaccea3b61455705286ede0de239dd84406352bb/src/opentimelineio/serialization.cpp#L570

Any ideas why this might be happening and how I can try going about it?

I don't get a segfault when I clone `itemClone1` instead, and for the third clone I use `item` again.
The error seems to be when I clone item twice consecutively.

Reply:
@jminor @meshula @ssteinbach @reinecke

Reply:
_schema_name_for_reference() is return a reference to a string, so as long as value is in scope, schema_type_name should be live. If value itself has been deallocated you'd get the segfault. If you could write a tiny cpp program to demonstrate the bug, or tell me how to reproduce it in an existing test, I'd be happy to take a look at it.

Reply:
You should be able to reproduce it using COTIOCompositionTests.

The procedure would be:

- clone the repo
- `cd OpenTimelineIO/src/c-opentimelineio`
- `mkdir build`
- `cd build`
- `cmake -DCMAKE_INSTALL_PREFIX=<install dir> ..`
- `make`

Then to run all tests:
- `make test`

But I think here it would be better to just run the composition tests:

Inside the build directory:

- `./tests/COTIOCompositionTests`

With the current code you should get a segfault at line 947 in `COTIOCompositionTests.cpp`.

I put breakpoints before and after both the clone operations which led me to the name for reference function.

If you comment out the second clone operation, you shouldn't get the segfault.

Reply:
@meshula did you get a chance to take a look?